src/pcp2openmetrics/pcp2openmetrics.py

```diff
@@ -0,0 +1,569 @@
+            else: 
+                subset = ['domainname', 'test', 'groupid', 'hostname', 'machineid', 'userid', 'instname', 'instid', 'agent', 'device_type', 'indom_name']
+            for i, key in enumerate(subset):
+                if key not in new_dict: continue
+                if i != 0: result += ','
+                result += '%s="%s"' % (key, new_dict[key])
+            return '{' + result + '}'
+
+        results = self.pmconfig.get_ranked_results(valid_only=True)
+
+        body = ''
+        for metric in results:
+            body += "Metric %s details (last fetch: %d):\n" % (metric, ts)
+            i = list(self.metrics.keys()).index(metric)
+            desc = self.pmconfig.descs[i]
+            context = self.pmfg.get_context()
+            pmid = context.pmLookupName(metric)
+
+            body += '# HELP %s %s\n' % (openmetrics_name(metric), openmetrics_help(context, pmid))
+            body += '# TYPE %s %s\n' % (openmetrics_name(metric), openmetrics_type(context, pmid))
+            
+            for inst, name, value in results[metric]:
+                if isinstance(value, float):
+                    fmt = "." + str(self.metrics[metric][6]) + "f"
+                    value = format(value, fmt)
+                elif isinstance (value, int):
+                    fmt = "d"
+                    value = format(value, fmt)
+                else:
+                    str(value)
+                body += '%s%s %s\n' % (openmetrics_name(metric), openmetrics_labels(metric, context, inst, name, desc), value)
+        
+        if self.url:
+            auth = None
+            if self.http_user and self.http_pass:
+                auth = requests.auth.HTTPBasicAuth(self.http_user, self.http_pass)
+            try:
+                timeout = self.http_timeout
+                headers = {'Content-Type': 'application/openmetrics-text'}
+                res = requests.post(self.url, data=body, auth=auth, headers=headers, timeout=timeout)
+                if res.status_code > 299:
+                    msg = "Cannot send metrics: HTTP code %s\n" % str(res.status_code)
+                    sys.stderr.write(msg)
+            except requests.exceptions.ConnectionError as post_error:
+                msg = "Cannot connect to server at %s: %s\n" % (self.url, str(post_error))
+                sys.stderr.write(msg)
+            except Exception as error:
+                msg = "HTTP Connection error"
+                sys.stderr.write(msg)
+        else:
+            print(body)
+
+    def finalize(self):
+        """ Finalize and clean up """
+        if self.writer:
+            try:
+                self.writer.write("\n")
+                self.writer.flush()
+            except IOError as write_error:
+                if write_error.errno != errno.EPIPE:
+                    raise
+            try:
+                self.writer.close()
+            except Exception:
+                pass
```

Comment:
Hmm, that is a pretty broad exception handler - what is that ignoring in particular?